ls1intum/Artemis
# `Programming exercises`: Change default branch from master to main

Changes the default branch from `master` to `main` and makes sure to use the configuration in the yml files in the production and test code.

**Note:** Administrators can override the yml configuration with a different default branch and could still fall back to `master` if needed.

### Checklist
#### General


#### Server
- [x] I adapted multiple integration tests (Spring) related to the features (with a high test coverage).
#### Changes affecting Programming Exercises

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Create a new programming exercise (instructor) and participate (student). Make sure the `main` branch is used
2. Student: Participate in an existing programming exercise (in which master was used) and make sure everything still works
3. Instructor: Import an existing programming exercise (in which master was used) and make sure everything works (also when participating as a student)

Note: this should be tested on both environments (Bitbucket/Bamboo and Gitlab/Jenkins)

### Review Progress





#### Code Review
#### Manual Tests
